lib/platform/github/index.ts

```diff
@@ -1719,3 +1725,72 @@ export async function getVulnerabilityAlerts(): Promise<VulnerabilityAlert[]> {
   }
   return alerts;
 }
+
+async function pushFiles(
+  { branchName, message }: CommitFilesConfig,
+  { parentCommitSha, commitSha }: CommitResult
+): Promise<CommitSha | null> {
+  try {
+    // Push the commit to GitHub using a custom ref
+    // The associated blobs will be pushed automatically
+    await pushCommitToRenovateRef(commitSha, branchName);
+    // Get all the blobs which the commit/tree points to
+    // The blob SHAs will be the same locally as on GitHub
+    const treeItems = await listCommitTree(commitSha);
+
+    // For reasons unknown, we need to recreate our tree+commit on GitHub
+    // Attempting to reuse the tree or commit SHA we pushed does not work
+    const treeRes = await githubApi.postJson<{ sha: string }>(
+      `/repos/${config.repository}/git/trees`,
+      { body: { tree: treeItems } }
+    );
+    const treeSha = treeRes.body.sha;
+
+    // Now we recreate the commit using the tree we recreated the step before
+    const commitRes = await githubApi.postJson<{ sha: string }>(
+      `/repos/${config.repository}/git/commits`,
+      { body: { message, tree: treeSha, parents: [parentCommitSha] } }
+    );
+    const remoteCommitSha = commitRes.body.sha;
+
+    // Create branch if it didn't already exist, update it otherwise
+    if (git.branchExists(branchName)) {
+      // This is the equivalent of a git force push
+      // We are using this REST API because the GraphQL API doesn't support force push
+      await githubApi.patchJson(
+        `/repos/${config.repository}/git/refs/heads/${branchName}`,
+        { body: { sha: remoteCommitSha, force: true } }
+      );
+    } else {
+      await githubApi.postJson(`/repos/${config.repository}/git/refs`, {
+        body: { ref: `refs/heads/${branchName}`, sha: remoteCommitSha },
+      });
+    }
+
+    return remoteCommitSha;
+  } catch (err) {
+    logger.debug({ branchName, err }, 'Platform-native commit: unknown error');
+    return null;
+  }
+}
+
+export async function commitFiles(
+  config: CommitFilesConfig
+): Promise<CommitSha | null> {
+  const commitResult = await git.prepareCommit(config); // Commit locally and don't push
+  if (!commitResult) {
+    const { branchName, files } = config;
+    logger.debug(
+      { branchName, files: files.map(({ path }) => path) },
+      `Platform-native commit: unable to prepare for commit`
+    );
+    return null;
+  }
+  // Perform the commits using REST API
+  const pushResult = await pushFiles(config, commitResult);
+  if (!pushResult) {
+    return null;
+  }
+  // Because the branch commit was done remotely via REST API, now we git fetch it locally
```

Comment:
```suggestion
  // Because the branch commit was done remotely via REST API, now we git fetch it locally.
  // We also do this step when committing/pushing using local git tooling.
```